### PR TITLE
Escape user strings

### LINKED ISSUE #1592: Unescaped HTML tags in names
Using HTML tags in names for coordinates etc. can inject those tags into the notebook HTML:
```.py
a = sc.DataArray(data=sc.Variable(['<a>x'], values=[0, 1, 2.]),
                 coords={'<a>x': sc.Variable(['<a>x'], values=[0.1, 0.2, 0.3])})
```
- `sc.table` injects the tag:
tagtable
If you inject a valid link here, it is clickable in the table.

- `sc.show` breaks without output if the HTML is invalid (adding a closing tag shows the boxes as normal but also produces a usable link)
tagshow

- String output does not inject the HTML but it replaces some characters:
tagstr

- Plotting works fine and the normal operations all seem possible.

I don't think this is particularly important right now as all users are writing software on their own computers. But if scipp gets deployed on a server, this might be a security risk.

### PR DESCRIPTION
**Pull request by jl-wynen**
Fixes #1592 

I _think_ I got everything but I'm not completely sure!

### CODE REVIEW

**Review by SimonHeybrock — APPROVED** [review]

File: python/src/scipp/plot/widgets.py [hunk 1]
```diff
@@ -156,7 +158,7 @@ def _add_masks_controls(self, masks):
                 for key in masks[name]["names"]:
                     self.mask_checkboxes[name][key] = ipw.Checkbox(
                         value=masks[name]["names"][key],
-                        description="{}:{}".format(name, key),
+                        description="{}:{}".format(escape(name), escape(key)),
```
Comment (nvaytet) [inline python/src/scipp/plot/widgets.py:161]: I'm not sure how you decide what needs escaping and what doesn't, but there are the dimension buttons (around L104) that have the dim labels as description, maybe this also needs escaping?

Then I guess you may also need to look at `panel1d.py`.

**Review by nvaytet — CHANGES_REQUESTED** [review]

File: python/src/scipp/plot/widgets.py (hunk 1 above)
Comment (jl-wynen) [reply python/src/scipp/plot/widgets.py:161]: I decided by trying out all sorts of outputs.

Are you referring to this? https://github.com/scipp/scipp/blob/2290ddeda0d5388a4261f08cf50f969250d306c7/python/src/scipp/plot/widgets.py#L106

It looks like the widgets escape the string for us:
plot_buttons

The figures all seem to work properly, too. At least I didn't find an example where it broke.

Comment (nvaytet) [reply python/src/scipp/plot/widgets.py:161]: Yes this is what i was talking about.
I tried a few more things, but I think what you got it all (weird that the checkboxes descriptions need special handling)

**Review by nvaytet — APPROVED** [review]

File: python/src/scipp/plot/widgets.py (hunk 1 above)
Comment (jl-wynen) [reply python/src/scipp/plot/widgets.py:161]: Well, they are formatted by ipywidgets. Our problems were all with places where we constructed HTML manually.

**Comment by SimonHeybrock** [comment]
@jl-wynen There are some conflicts.

**Comment by jl-wynen** [comment]
No, there wasn't oO

**Pull request merged**